code/controllers/subsystem/timer.dm

```diff
@@ -68,9 +76,37 @@ var/datum/subsystem/timer/SStimer
 		event.argList = args.Copy(5)
 
 	// Check for dupes if unique = 1.
-	if(unique)
-		if(event.hash in SStimer.hashes)
-			return
+	switch(unique)
+		if(TIMER_OLDEST) // Uses the first timer that was created.
+			if(event.hash in SStimer.unique)
+				qdel(src)
+				return
+			SStimer.unique[event.hash] = event
+		if(TIMER_NEWEST) // Uses the most recently created timer.
+			var/datum/timedevent/old = SStimer.unique[event.hash]
+			if(old)
+				qdel(old)
+			SStimer.processing -= old // In case qdel doesnt get to it fast enough to remove it from the processing list.
```

Comment:
but it wont cause an error, either. I was concerned that trying to qdel(null) might be an issue. And for the later ones, trying to access null.vars might be an issue.

removing null from a list isnt an issue. However removing it should probably be done BEFORE qdelling it.